src/Models/ObjectMap/Entities/Associations/AssociationStubBase.php

```diff
@@ -49,6 +52,56 @@ abstract class AssociationStubBase
      */
     protected $baseType;
 
+    /**
+     * Any assocations this stub is a member of.
+     *
+     * @var Association[]
+     */
+    protected $associations = [];
+
+    /**
+     * @var EntityGubbins the entity this stub lives on.
+     */
+    protected $entity;
+
+    /**
+     * Sets the entity owning this AssocationStub
+     *
+     * @param EntityGubbins $entity
+     */
+    public function setEntity(EntityGubbins $entity): void
+    {
+        $this->entity = $entity;
+    }
+
+    /**
+     * Gets the entity owning this AssocationStub
+     *
+     * @return EntityGubbins
+     */
+    public function getEntity(): EntityGubbins
+    {
+        return $this->entity;
+    }
+    /**
+     * Adds this stub as a member of an assocation
+     *
+     * @param Association $newAssocation the new assocation to be a member of
+     */
+    public function addAssociation(Association $newAssocation): void {
+        $this->associations[spl_object_hash($newAssocation)] = $newAssocation;
```

Comment:
Maybe a comment to explain why you're using spl_object_hash to use an object as a key